src/components/calculator.tsx

```diff
@@ -0,0 +1,33 @@
+import DigitButton from '@components/digitButton'
+import OperationButton from '@components/operationButton'
+
+import { DIGIT_NUMBER, OPERATIONS } from '@/constant/calculator'
+
+import useCalculator from '@/customs/useCalculator'
+
+const Calculator = () => {
+  const { total, handleDigit, handleOperation, handleReset } = useCalculator()
+
+  return (
+    <div className="calculator">
+      <h1 id="total">{total}</h1>
+      <div className="digits flex">
+        {DIGIT_NUMBER.map((digit) => (
+          <DigitButton key={digit} digit={digit} onClick={handleDigit} />
+        ))}
+      </div>
+      <div className="modifiers subgrid">
+        <button className="modifier" onClick={handleReset}>
+          AC
+        </button>
+      </div>
+      <div className="operations subgrid">
+        {OPERATIONS.map((operation) => (
+          <OperationButton key={operation} operation={operation} onClick={handleOperation} />
+        ))}
+      </div>
```

Comment:
`DigitButton`과 `OperationButton`은 하는 일이 큰 차이가 없는거 같은데요, 
요러한 도메인을 구분으로 컴포넌트를 구분하는 방법도 많이 쓰이지만,
도메인과 완전히 프리한 컴포넌트를 구성한 다음에 이를 재활용하는 방법도 많이 쓰입니다 😄 


```suggestion
      <div className="digits flex">
        {DIGIT_NUMBER.map((digit) => (
          <Button className="digit" key={digit} onClick={() => handleDigit(digit)}>
            {digit}
          </Button>
        ))}
      </div>
      <div className="modifiers subgrid">
        <button className="modifier" onClick={handleReset}>
          AC
        </button>
      </div>
      <div className="operations subgrid">
        {OPERATIONS.map((operation) => (
          <Button className="operation" key={operation} onClick={() => handleOperation(operation)} />
  {operation}
         </Button> 
        ))}
      </div>
```


요렇게 보면 `DigitButton`과 `OperationButton` 모두 똑같지 않나요?

버튼 공통의 스타일이나 기능이 있다면 Button 컴포넌트에서 구현하구요 😄

Reply:
두 가지 다른 버튼이 아니라 하나의 계산기 버튼으로 통합하니 더 깔끔해진 것 같아요. 감사합니다!